### PR TITLE
Discard op code m change events

### LINKED ISSUE #41647: Debezium heartbeat not identified correctly
### Connector Name

source-postgres

### Connector Version

3.4.24

### What step the error happened?

None

### Relevant information

This was found in relation to #41622.
The way we identify debezium heartbeat messages and omit them from getting processed as change events is broken.
We essentially identify as a heartbeat incoming debezium events which contain no `source` node.
The internal format of heartbeats may have changed and after applying the fix in #41622 on an affected workspace, the added logs show a heartbeat packet which we attempt to process as a change event. causing an error/
See example

### Relevant log output

```shell
{"op":"m","ts_ms":1720724527278,"source":{"version":"2.6.2.Final","connector":"postgresql","name":"xxxx","ts_ms":1720670367584,"snapshot":null,"db":"xxxx","sequence":"[\"7360842577240\",\"7360842578440\"]","ts_us":1720670367584260,"ts_ns":1720670367584260000,"schema":"","table":"","txId":3570146,"lsn":7360842578440,"xmin":null},"message":{"prefix":"datastream","content":"Y2RjIGhlYXJ0YmVhdA=="}}
```
Note a `source` exists.
The value of `content` is base64 encoded "cdc heartbeat".

The correct way to deal with it is probably filtering out `"op": "m"` events, signifying a debezium [message event](https://debezium.io/documentation/reference/stable/connectors/postgresql.html#postgresql-message-events), as opposed to a change event.

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by rodireich**
## What
Fixes https://github.com/airbytehq/airbyte/issues/41647
Debezium change events are coming with an operation code.
Op code "m" signifies a "message" event that is irrelevant for syncing as it contains no before or after data.
In this case a 3rd party wrote arbitrary messages into the WAL log which caused us to fail parsing.
We want to only handle known event types.

## How
Discard debezium change events that has `op: m`.

## User Impact
Should not impact users as it should unblock cases a message event breaks us.

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/debezium/internals/DebeziumRecordIterator.kt [hunk 1]
```diff
@@ -171,6 +171,13 @@ class DebeziumRecordIterator<T>(
             }
 
             val changeEventWithMetadata = ChangeEventWithMetadata(next)
+
+            // #41647: discard event type with op code 'm'
+            if (!isEventTypeHandled(changeEventWithMetadata)) {
+                LOGGER.info { "Event type not handled: $next" }
```
Comment (evantahler) [inline airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/debezium/internals/DebeziumRecordIterator.kt:177]:
```suggestion
                LOGGER.info { "WAL event type not handled: $next" }
```

**Review by evantahler — APPROVED** [review]
Nice. 

It looks like other DBs can have the same issue, but we can republish those later - Postgres has the issue now.

File: airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/debezium/internals/DebeziumRecordIterator.kt (hunk 1 above)
Comment (rodireich) [reply airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/debezium/internals/DebeziumRecordIterator.kt]: Yes. It's likely that other DBs have similar use cases as in postgres. Thanks

**Pull request merged**